modules/python/test/test_misc.py

```diff
@@ -50,39 +68,538 @@ class Arguments(NewOpenCVTests):
         self.assertEqual(res2_1, "InputArrayOfArrays: empty()=false kind=0x00050000 flags=0x01050000 total(-1)=2 dims(-1)=1 size(-1)=2x1 type(0)=CV_64FC1 dims(0)=2 size(0)=1x4 type(0)=CV_64FC1")
         res2_2 = cv.utils.dumpInputArrayOfArrays([1.5])
         self.assertEqual(res2_2, "InputArrayOfArrays: empty()=false kind=0x00050000 flags=0x01050000 total(-1)=1 dims(-1)=1 size(-1)=1x1 type(0)=CV_64FC1 dims(0)=2 size(0)=1x4 type(0)=CV_64FC1")
-        a = np.array([[1,2],[3,4],[5,6]])
-        b = np.array([[1,2,3],[4,5,6],[7,8,9]])
+        a = np.array([[1, 2], [3, 4], [5, 6]])
+        b = np.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
         res3 = cv.utils.dumpInputArrayOfArrays([a, b])
         self.assertEqual(res3, "InputArrayOfArrays: empty()=false kind=0x00050000 flags=0x01050000 total(-1)=2 dims(-1)=1 size(-1)=2x1 type(0)=CV_32SC1 dims(0)=2 size(0)=2x3 type(0)=CV_32SC1")
-        c = np.array([[[1,2],[3,4],[5,6]]], dtype='f')
+        c = np.array([[[1, 2], [3, 4], [5, 6]]], dtype='f')
         res4 = cv.utils.dumpInputArrayOfArrays([c, a, b])
         self.assertEqual(res4, "InputArrayOfArrays: empty()=false kind=0x00050000 flags=0x01050000 total(-1)=3 dims(-1)=1 size(-1)=3x1 type(0)=CV_32FC2 dims(0)=2 size(0)=3x1 type(0)=CV_32FC2")
 
+    def test_parse_to_bool(self):
+        def try_to_convert(value):
+            try:
+                result = cv.utils.dumpBool(value).lower()
+            except Exception as e:
+                self.fail(
+                    '{} "{}" is risen for conversion {} of type {}'.format(
+                        type(e).__name__, e, value, type(value).__name__
+                    )
+                )
+            else:
+                return result
+
+        for convertible_true in (True, 1, 64, np.bool(1), np.int8(123),
+                                 np.int16(11), np.int32(2), np.int64(1)):
+            actual = try_to_convert(convertible_true)
+            self.assertEqual('bool: true', actual,
+                             msg="Can't convert {0} of type {1}\n"
+                             "Expected: 'bool: true' vs Actual: {2}".format(
+                                 convertible_true,
+                                 type(convertible_true).__name__,
+                                 actual
+                             ))
+
+        for convertible_false in (False, 0, np.uint8(0)):
+            actual = try_to_convert(convertible_false)
+            self.assertEqual('bool: false', actual,
+                             msg="Can't convert {0} of type {1}\n"
+                             "Expected: 'bool: false' vs Actual: {2}".format(
+                                 convertible_false,
+                                 type(convertible_false).__name__,
+                                 actual
+                             ))
+
+        a = np.array([1, 2, 0], dtype=np.bool)
+        for i in range(len(a)):
```

Comment:
Is it necessary to check each value in the array, can it be moved to previous blocks?
```.py
for convertible_true in (..., np.array([1, 2, 0], dtype=np.bool)[1]):
...
for convertible_false in (..., np.array([1, 2, 0], dtype=np.bool)[2]):
```

Same comment for all following test cases.

Reply:
If this check will be moved to "convertible" block, it won't be possible to determine if conversion error occurs when array was indexed or just normal check.

Reply:
It can be determined from current iteration index.

Reply:
Hmm, I think we can completly eliminate this check if we add array element type to appropriate list. E.G:
```python
type(np.array([1, 2], dtype=np.bool)[0]) # yields to np.bool_
type(np.array([1, 2], dtype=ctypes.c_size_t)[0]) # yields to np.uint64
type(np.array(['t1', 't2'])[0]) # yields to np.str_ which is np.str
```